src/main/kotlin/org/vineflower/ijplugin/VineflowerSettingsPanel.kt

```diff
@@ -70,83 +60,36 @@ class VineflowerSettingsPanel(var prevVineflowerVersion: SemVer?) {
             vineflowerSettingsPanel.repaint()
             return
         }
-        val classLoader = VineflowerState.getInstance().getVineflowerClassLoader().getNow(null) ?: return
-        val preferencesClass = classLoader.loadClass("org.jetbrains.java.decompiler.main.extern.IFernflowerPreferences")
 
-        val fieldAnnotations = VineflowerPreferences.FieldAnnotations(classLoader)
+        val preferences = VineflowerPreferences.create()
+        val allSettings = mutableListOf<VineflowerPreferences.SettingsEntry>()
+        preferences.setupSettings(allSettings, vineflowerSettings)
 
-        @Suppress("UNCHECKED_CAST")
-        val defaults = (preferencesClass.getField("DEFAULTS").get(null) as Map<String, *>).toMutableMap()
-        defaults.putAll(VineflowerPreferences.defaultOverrides)
+        val groups = allSettings.groupBy { it.group }
+        val keys = groups.keys.sortedBy { it }
```

Comment:
You can just group to a sorted set:
```suggestion
        val groups = allSettings.groupByTo(TreeSet()) { it.group }
```